### ISSUE TITLE
html.Encode in titles and other texts

### ISSUE THREAD

**Issue by bo-bac** [nopSolutions/nopCommerce#5675]
nopCommerce version: all

Steps to reproduce the problem: 
1 Install site use any Cyrillic language
2 Set page title for Home page or any
3 Go to that page and View page source
4 There will be something like this
`<title>WD1. SUBARUMIR - &#x437;&#x430;&#x43F;&#x447;&#x430;&#x441;&#x442;&#x438; &#x432; &#x43D;&#x430;&#x43B;&#x438;&#x447;&#x438;&#x438;</title>`

Expected
It seems we need not any additional encoring.

The same in several places over the code as well.

**Comment by AndreiMaz** [comment]
@bo-bac  It works fine both ways (decoded correctly in browser). But we'll check anyway. Thanks for the suggestion

**Comment by bo-bac** [comment]
yes it is decoded. but please use 'View page source' in the browser.

https://user-images.githubusercontent.com/4517566/117577292-de699380-b0f1-11eb-851d-737f52a56b2f.mp4

**Comment by RomanovM** [comment]
I don't think there is an issue with encoding, anyway it's correctly decoded by robots and browsers, see the previous discussion about this in #4115

**Comment by bo-bac** [comment]
It's clear that in a such way we avoid vulnerability problems.
But along with potential javascript issues it is definitely increases page size.
Too heavy pages imho are not good for SEO even if bots know how to decode content.
Or who cares today about page sizes?)

What if use a kind of html.AttributeEncode (encode only & <)?

**Comment by bo-bac** [comment]
plus as I can see there is no consistency today
```
<div class="product-name">
  <h1>
         &#x424;&#x43E;&#x442;&#x43E;&#x43A;&#x430;&#x43C;&#x435;&#x440;&#x430; Nikon D5500 DSLR &quot; &amp; &lt; \
  </h1>
  </div>
  <div class="short-description">
        Легка, крута фотокамера Nikon D5500 з поворотним тачскріном
  </div>
```

**Comment by bo-bac** [comment]
here is two points to consider
1
it is possible to configure Html.Encode so it will allow not only BasicLatin chars to be unescaped
will fix #5675 if add to appsettings.json

"XssProtectionConfig": {
    "TextEncoderAllowedRanges": [
      "UnicodeRanges.All"
    ]
  }
discussed here aspnet/HttpAbstractions#315
and the same fix here OrchardCMS/OrchardCore#4041

2
is seems we should use HTML Sanitation for rich HTML Input in such places like this
@Html.Raw(Model.FullDescription)
cuz today it possible to save something like <script>alert('qqq')</script> and it will run
this one for example https://github.com/mganss/HtmlSanitizer

**Comment by exileDev** [comment]
After some discus, we've decided not to implement this functionality out of the box for near versions. We're planning to consideration HTML-sanitizing. We'll open a new issue, if necessary.

Closed #5675
